medusa/cassandra_utils.py

```diff
@@ -238,45 +239,57 @@ def saved_caches_directory(self):
 
     @property
     def listen_address(self):
-        if 'listen_address' in self._config:
-            if self._config['listen_address']:
-                return self._config['listen_address']
+        if 'listen_address' in self._config and self._config['listen_address']:
+            return self._config['listen_address']
 
         return socket.gethostbyname(socket.getfqdn())
 
     @property
     def storage_port(self):
-        if 'storage_port' in self._config:
-            if self._config['storage_port']:
-                return self._config['storage_port']
-        return "7000"
+
+        internode_encrypt = self._config['server_encryption_options']['internode_encryption']
+        if internode_encrypt in self._config and internode_encrypt and internode_encrypt != "none":
+            ssl_storage_port = self._config['ssl_storage_port']
+            if ssl_storage_port in self._config and ssl_storage_port:
+                return ssl_storage_port
+            return "7001"
+        else:
+            storage_port = self._config['storage_port']
+            if storage_port in self._config and storage_port:
+                return storage_port
+            return "7000"
 
     @property
     def native_port(self):
-        if 'native_transport_port' in self._config:
-            if self._config['native_transport_port']:
-                return self._config['native_transport_port']
+
+        client_encryption = self._config['client_encryption_options']['enabled']
+        if client_encryption in self._config and client_encryption and client_encryption is True:
+            native_transport_port_ssl = self._config['native_transport_port_ssl']
+            if native_transport_port_ssl in self._config and native_transport_port_ssl:
+                return native_transport_port_ssl
+            return "9142"
+
+        native_transport_port = self._config['native_transport_port']
+        if native_transport_port in self._config and native_transport_port:
+            return native_transport_port
         return "9042"
 
     @property
     def rpc_port(self):
-        if 'rpc_port' in self._config:
-            if self._config['rpc_port']:
-                return self._config['rpc_port']
+        if 'rpc_port' in self._config and self._config['rpc_port']:
+            return self._config['rpc_port']
         return "9160"
 
     @property
     def seeds(self):
         seeds = list()
-        if 'seed_provider' in self._config:
-            if self._config['seed_provider']:
-                if self._config['seed_provider'][0]['class_name'].endswith('SimpleSeedProvider'):
-                    return self._config.get('seed_provider')[0]['parameters'][0]['seeds'].replace(' ', '').split(',')
+        if 'seed_provider' in self._config and self._config['seed_provider'] and \
```

Comment:
neat 👍 

I think we were afraid that unlike Java, Python would evaluate all the conditions at once, which could trigger an exception if the key was missing. Good to know it's not the case!

Reply:
Verified that this is still the case (exception for missing key) unless there is another option, I will leave this approach in place.